docs/concepts/get-offline-features.md

```diff
@@ -0,0 +1,85 @@
+
+```python
+user_feature_query = FeatureQuery(
+    feature_list=["feature_user_age",
+                  "feature_user_tax_rate",
+                  "feature_user_gift_card_balance",
+                  "feature_user_has_valid_credit_card",
+                  "feature_user_total_purchase_in_90days",
+                  "feature_user_purchasing_power"
+                  ],
+    key=user_id)
+
+product_feature_query = FeatureQuery(
+    feature_list=[
+                  "feature_product_quantity",
+                  "feature_product_price"
+                  ],
+    key=product_id)
+```
+
+And specify the location for the observation data:
+
+```python
+settings = ObservationSettings(
+    observation_path="wasbs://public@azurefeathrstorage.blob.core.windows.net/sample_data/product_recommendation_sample/user_observation_mock_data.csv",
+    event_timestamp_column="event_timestamp",
+    timestamp_format="yyyy-MM-dd")
+```
+
+And finally, specify the feature query and finally trigger the computation:
+
+```python
+client.get_offline_features(observation_settings=settings,
+                            feature_query=[user_feature_query, product_feature_query],
+                            output_path=output_path)
+
+```
+
+More details for the above APIs can be read from:
+
+- [ObservationSettings API doc](https://feathr.readthedocs.io/en/latest/feathr.html#feathr.ObservationSettings)
+- [client.get_offline_feature API doc](https://feathr.readthedocs.io/en/latest/feathr.html#feathr.FeathrClient.get_offline_features)
+
+## More on `Observation data`
+
+The path of a dataset as the 'spine' for the to-be-created training dataset. We call this input 'spine' dataset the 'observation' dataset. Typically, each row of the observation data contains:
+
+1. **Entity ID Column:** Column(s) representing entity id(s), which will be used as the join key to query feature value.
+
+2. **Timestamp Column:** A column representing the event time of the row. By default, Feathr will make sure the feature values queried have a timestamp earlier than the timestamp in observation data, ensuring no data leakage in the resulting training dataset. Refer to [Point in time Joins](./point-in-time-join.md) for more details.
+
+3. **Other columns** will be simply pass through to the output training dataset, which can be treated as immutable columns.
+
+## More on `Feature Query`
+
+After you have defined all the features, you probably don't want to use all of them in this particular program. In this case, instead of putting every features in this `FeatureQuery` part, you can just put a selected list of features. Note that they have to be of the same key.
+
+## Difference between `materialize_features` and `get_offline_features` API
```

Comment:
One more talking point should get mentioned here is regards these two api used in two different stage of feature engineering pipeline serves different purpose.

get_offline_features is to inference model trainings via offline store, while materialize_features is to inference mode servicing via online store with pre-computed features.

Reply:
Good point, added